### PR TITLE
Separate supplemental resources from materials section 

### LINKED ISSUE #570: separate supplemental resources from required materials. (multiple assignees encouraged)
On one side, some apprentices think that they need to throughly look through every material, and are getting way behind because of it.  On the other, many well-meaning volunteers have thrown the kitchen sink into the "materials" section, when many or even all of them should be considered "supplemental resources" (supplemental meaning additional, extra, not required). 

Look through topic outlines, and change one so that:
- the "slides" are part of the "materials" section
- the "lesson" explicitly directs how to use the mandatory "materials" (ex: "Read through [these slides]() )
- optional materials are under a "supplemental resources" section.
- if you finish, or notice a topic outline looks fine, please check it off on this list or let a Techtonica admin know it's complete.

Topic outlines whose materials sections not been reviewed:
- [x] https://github.com/Techtonica/curriculum/blob/master/professionalism/professionalism.md
- [x] https://github.com/Techtonica/curriculum/blob/master/ergonomics/ergonomics.md

- [x] https://github.com/Techtonica/curriculum/blob/master/command-line/vim.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/html/html.md

- [x] https://github.com/Techtonica/curriculum/blob/master/git-version-control/gitignore.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/css/css.md

- [x] https://github.com/Techtonica/curriculum/blob/master/css/media-queries.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/ui-ux-design/ui-ux-design.md
- [x] https://github.com/Techtonica/curriculum/blob/master/web-patterns/web-patterns.md
- [x] https://github.com/Techtonica/curriculum/blob/master/bootstrap/bootstrap.md
- [x] https://github.com/Techtonica/curriculum/blob/master/professionalism/building-confidence.md

- [x] https://github.com/Techtonica/curriculum/blob/master/github-collaboration/github-collaboration.md
- [x] https://github.com/Techtonica/curriculum/blob/master/reading-and-writing-documentation/documentation.md
- [x] https://github.com/Techtonica/curriculum/blob/master/deploying/deploying-1-personal-site.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-1.md
- [x] https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-2.md

- [x] https://github.com/Techtonica/curriculum/blob/master/runtime-complexity/runtime-complexity.md
- [x] https://github.com/Techtonica/curriculum/blob/master/testing-and-tdd/jasmine-testing.md
- [x] https://github.com/Techtonica/curriculum/blob/master/debugging/debugging.md
- [x] https://github.com/Techtonica/curriculum/blob/master/databases/relational-databases.md
- [x] https://github.com/Techtonica/curriculum/blob/master/react-js/react.md
- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/intro-to-data-structures.md
- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/stack.md
- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/deque.md

- [x] https://github.com/Techtonica/curriculum/blob/master/data-structures/trees.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/algorithms/intro-to-algorithms.md
- [x] https://github.com/Techtonica/curriculum/blob/master/algorithms/searching.md
- [x] https://github.com/Techtonica/curriculum/blob/master/algorithms/sorting.md

- [x] https://github.com/Techtonica/curriculum/blob/master/open-source/open-source.md
 
- [x] https://github.com/Techtonica/curriculum/blob/master/roles-in-tech/machine-learning.md

### PR DESCRIPTION
**Pull request by vsharma20v**
Separate supplemental resources from the materials section in memoization.md file and also fixed some typo errors & white spaces.
Issue Addressed #570

### CODE REVIEW

File: runtime-complexity/memoization.md [hunk 1]
```diff
@@ -1,212 +1,120 @@
...
-
 - when new results are derived check to see if those results are in the hash table;
-
 - if so, call on the results from the hash table to save time, if not, add the new results to the hash table
 
 ### Materials
-
-  -  [Memoization Slideshow](https://docs.google.com/presentation/d/1BipDMgjZd3u-QsrPNCljH-Wv2l3tYRAUz8LWnxzt4s8/edit#slide=id.p)
-
+-  [Memoization Slideshow](https://docs.google.com/presentation/d/1BipDMgjZd3u-QsrPNCljH-Wv2l3tYRAUz8LWnxzt4s8/edit#slide=id.p)
```
Comment (Kundan28) [inline runtime-complexity/memoization.md]: @alodahl I don't have edit access for this slide. So I can't review it.

Comment (Kundan28) [reply runtime-complexity/memoization.md]: @vsharma20v Make sure that the slideshow follows all the formatting guidelines mentioned in #600

File: runtime-complexity/memoization.md [hunk 2]
```diff
@@ -1,212 +1,120 @@
...
 
-  
-
-Read through the history of memoization and 'What is memoization' slides and get familiar with the new vocabulary.
-
-  
-
-Before we read about Memoization,we need to understand that it is a part of Dynamic Programming. **Dynamic Programming** is a method for solving a complex problem by breaking it down into a collection of simpler subproblems, solving each of those subproblems just once, and storing their solutions using a memory-based data structure (array, map,etc). Dynamic programming is mainly an optimization over plain recursion. Wherever we see a recursive solution that has repeated calls for same inputs, we can optimize it using Dynamic Programming. The idea is to simply store the results of subproblems, so that we do not have to re-comupute them when needed later. Memoization (top-down approach), along with its “sibling” tabulation (bottom-up approach), are the two main Dynamic Programming techniques.
+Read through the history of Memoization and 'What is memoization' slides and get familiar with the new vocabulary.
```
Comment (Kundan28) [inline runtime-complexity/memoization.md]: slides -> slideshow

File: runtime-complexity/memoization.md [hunk 3]
```diff
@@ -1,212 +1,120 @@
...
 Notice the difference in Time Complexity between using recursion and a while loop.
 
-  
-
-Finally look at the example code of the Fibonacci Sequence using memoization on slide 13 and again notice the Time Complexity and Space Complexity.
-
-  
+Finally, look at the example code of the Fibonacci Sequence using memoization on slide 13 and again notice the Time Complexity and Space Complexity.
 
 Take a look at the comparison chart showing runtime complexity and space complexity for different sample size. what is your biggest take away from slides 17 and 18?
```
Comment (Kundan28) [inline runtime-complexity/memoization.md]: what -> What

File: runtime-complexity/memoization.md [hunk 4]
```diff
@@ -1,212 +1,120 @@
...
     Ans:
-```  function fibonacci(num, memo)```
+    ```function fibonacci(num, memo)```
 
 
  2. How do we define 'memo' in the function which was not defined when we used recursion?
-
     What would you add to the function?
 
     Ans: create a hash table to store the data to be used again when called upon.
```
Comment (Kundan28) [inline runtime-complexity/memoization.md]: create -> Create

Comment (alodahl) [reply runtime-complexity/memoization.md]: Not his job for #570, but thanks for the vigilance Kundan 😊

**Pull request merged**